core-bundle/src/EventListener/DataContainer/PreviewLinkListener.php

```diff
@@ -112,7 +112,9 @@ public function createFromUrl(DataContainer $dc): void
                 case 'toggle':
                 case 'delete':
                 default:
-                    if ($dc->activeRecord->createdBy !== $userId) {
+                    $createdBy = $this->connection->fetchOne('SELECT createdBy FROM tl_preview_link WHERE id=?', [$dc->id]);
+
+                    if ($createdBy !== $userId) {
```

Comment:
The strict comparison will not work here, because the DB result will be a string and `$userId` might be an integer.

Reply:
We should IMHO cast both the DB result as outlined [here](https://github.com/contao/contao/pull/3934/files#r785445810) and the assignment in L84:
 `$userId = (int) ($user instanceof BackendUser ? $user->id : 0);`

Reply:
See 8116179273ebda6ffca7ab6b316992d6e98624ff